src/core/namespace.c

```diff
@@ -111,6 +114,7 @@ typedef struct MountEntry {
         LIST_HEAD(MountOptions, image_options_const);
         char **overlay_layers;
         VeritySettings verity;
+        ImageClass required_class; /* Used for live updates to skip unapplicable images */
```

Comment:
hmm, thinking about this, maybe call this "filter_class" here.

to me "require" sounds like "failing if doesn't apply", but here we mean "skip if doesn't apply".

(the lower-level paramater can remain as required_class, because if you fail there with ENOCSI, then that's a failure, hence filter would be the wrong name)